aten/src/ATen/Context.h

```diff
@@ -250,11 +243,17 @@ static inline bool hasMKLDNN() {
 }
 
 static inline void manual_seed(uint64_t seed) {
-  globalContext().defaultGenerator(DeviceType::CPU).manualSeed(seed);
+  auto& gen = detail::getDefaultCPUGenerator();
+  // See Note [Thread-safety and Generators]
+  std::lock_guard<std::mutex> lock(gen->mutex_);
+  gen->set_current_seed(seed);
   // NB: Sometimes we build with CUDA, but we don't have any GPUs
   // available. In that case, we must not seed CUDA; it will fail!
   if (hasCUDA() && detail::getCUDAHooks().getNumGPUs() > 0) {
-    globalContext().defaultGenerator(DeviceType::CUDA).manualSeedAll(seed);
+    globalContext().lazyInitCUDA();
+    auto& generator = globalContext().generator_registry[static_cast<int>(DeviceType::CUDA)];
+    AT_ASSERT(generator);
+    generator->manualSeedAll(seed);
```

Comment:
Just double checking: you inlined the old code for `defaultGenerator` here, is that right?

Reply:
Yes.